src/app/shared/services/api/record.repo.ts

```diff
@@ -1,14 +1,50 @@
+/* @format */
 import { RecordVO, FolderVO, RecordVOData, SimpleVO } from '@root/app/models';
-import { BaseResponse, BaseRepo, LeanWhitelist } from '@shared/services/api/base';
-import { map } from 'rxjs/operators';
-import { Observable } from 'rxjs';
-
+import {
+  BaseResponse,
+  BaseRepo,
+  LeanWhitelist,
+} from '@shared/services/api/base';
 
 import { StorageService } from '@shared/services/storage/storage.service';
 import { ThumbnailCache } from '@shared/utilities/thumbnail-cache/thumbnail-cache';
+import { getFirst } from '../http-v2/http-v2.service';
+
+const MIN_WHITELIST: (keyof RecordVO)[] = [
+  'recordId',
+  'archiveNbr',
+  'folder_linkId',
+];
+const DEFAULT_WHITELIST: (keyof RecordVO)[] = [
+  ...MIN_WHITELIST,
+  'displayName',
+  'description',
+  'displayDT',
+];
+
+class MultipartUploadUrlsList {
+  public urls: string[] = [];
+  public uploadId: string;
+  public key: string;
 
-const MIN_WHITELIST: (keyof RecordVO)[] = ['recordId', 'archiveNbr', 'folder_linkId'];
-const DEFAULT_WHITELIST: (keyof RecordVO)[] = [...MIN_WHITELIST, 'displayName', 'description', 'displayDT'];
+  protected isInstance(obj: unknown): obj is MultipartUploadUrlsList {
```

Comment:
I am just curious, what is the purpose of this? Obviously it cheks those keys, but why?

Reply:
can there be instances when those keys are other type? Or just a safeguard?

Reply:
This is a TypeScript type guard since we're getting an object of `unknown` type.